src/Documenter.jl

```diff
@@ -511,19 +511,23 @@ function git_push(
                 if isempty(tag)
                     gitrm_copy(target_dir, latest_dir)
                     Writers.HTMLWriter.generate_siteinfo_file(latest_dir, "latest")
-                else
-                    gitrm_copy(target_dir, stable_dir)
-                    Writers.HTMLWriter.generate_siteinfo_file(stable_dir, "stable")
+                elseif occursin(Base.VERSION_REGEX, tag)
+                    version = VersionNumber(tag)
+                    # only push to stable if this is the latest stable release
+                    versions = filter!(x -> occursin(Base.VERSION_REGEX, x), readdir(dirname))
+                    maxver = mapreduce(x -> VersionNumber(x), max, v"0.0.0", versions)
+                    if version >= maxver && version.prerelease == () # don't deploy to stable for prereleases
+                        gitrm_copy(target_dir, stable_dir)
+                        Writers.HTMLWriter.generate_siteinfo_file(stable_dir, "stable")
+                    end
                     gitrm_copy(target_dir, tagged_dir)
                     Writers.HTMLWriter.generate_siteinfo_file(tagged_dir, tag)
-                    # Build a `release-*.*` folder as well when the travis tag is
-                    # valid, which it *should* always be anyway.
-                    if occursin(Base.VERSION_REGEX, tag)
-                        version = VersionNumber(tag)
-                        release = "release-$(version.major).$(version.minor)"
-                        gitrm_copy(target_dir, joinpath(dirname, release))
-                        Writers.HTMLWriter.generate_siteinfo_file(joinpath(dirname, release), release)
-                    end
+                    # Build a `release-*.*` folder as well
+                    release = "release-$(version.major).$(version.minor)"
+                    gitrm_copy(target_dir, joinpath(dirname, release))
+                    Writers.HTMLWriter.generate_siteinfo_file(joinpath(dirname, release), release)
+                else
+                    Compat.@warn("tag `$tag` not a valid VersionNumber")
```

Comment:
I warn already in the beginning of `deploydocs` though, where we do the other checks. Can keep this one as well though, for good measure -- it might help when debugging.